src/Core/Localization/Locale.php

```diff
@@ -149,6 +149,7 @@ public function formatNumber($number)
      */
     public function formatPrice($number, $currencyCode)
     {
+        if ( $number === null ) return null;
```

Comment:
@jolelievre Then the best solution would be to restrict the input number as `string` only. A simple casting from the calling code would resolve any issue.
I know this would cause a massive BC break, but at least we avoid any confusion.

Reply:
@saulaski an empty string throw an exception too, so casting not fix it 
I think the main question is, formating empty price must be throwing an exception ? 
In this case return exception is a way to detect unexpected behavior or a way to add unexpected behavior ?

Reply:
@mkdgs In this case I agree with @jolelievre, an invalid input to format has to throw an exception.
Number to format can be empty, or simply an invalid number string with unexpected characters. In both cases the value cannot be formatted as a valid number.
The calling code has to use the formatter only when appropriate.

Reply:
@saulaski, @jolelievre 
I think it's excessive for the formatPrice(), it's used for presentational purpose, not validation. 
It's only formating a price, and is not used for calculation.
If formating stuff is needed for validation i think it's a problem.
Imho i think formatPrice() is just a handly method must process empty string and null without any exception.  

If it's definitely not the best choice, I'll be sad.
But I will dry my tears, close this PR and fix that in the calling code.

Reply:
Only a question of point of view.
`formatPrice()` expects a valid value to format as price.
If any value should get formatted as price, or empty string if invalid, then a dedicated method is more appropriate: `formatValueAsPrice()`